dashboard/views.py

```diff
@@ -255,11 +253,61 @@ def grade_distribution(request, course_id=0):
     data = {
         "course_id": course_id
     }
-    eventlog(request.user, EVENT_VIEW_GRADE_DISTRIBUTION, extra=data)
+    eventlog(request.user, EventLogTypes.EVENT_VIEW_GRADE_DISTRIBUTION.value, extra=data)
 
     return HttpResponse(df.to_json(orient='records'))
 
 
+def update_user_default_selection_for_views(request, course_id=0):
+    logger.info(update_user_default_selection_for_views.__name__)
+    current_user = request.user.get_username()
+    default_selection = dict(request.GET)
+    logger.info('dic: {}'.format(default_selection))
+    default_type = [*default_selection][0]
+    default_type_value = default_selection.get(default_type)[0]
+    logger.info(f"request to set default for type: {default_type} and default_type value: {default_type_value}")
+
+    # json for eventlog
+    data = {
+        "course_id": course_id,
+        "default_type": default_type,
+        "default_value": default_type_value
+    }
+    eventlog(request.user, EventLogTypes.EVENT_VIEW_SET_DEFAULT.value, extra=data)
+    key = 'default'
+    try:
+        obj, create_or_update_bool = UserDefaultSelection.objects.set_user_defaults(course_id, current_user,
+                                                                                    default_type,
+                                                                                    default_type_value)
+        logger.info(
+            f"""setting default returns with success with response {obj.__dict__} and entry created or Updated: {create_or_update_bool}
+                        for user {current_user} in course {course_id} """)
+        value = 'success'
+    except (ObjectDoesNotExist, Exception) as e:
+        logger.info(f"updating default failed due to {e} for user {current_user} in course: {course_id} ")
+        value = 'fail'
+    return HttpResponse(json.dumps({key: value}))
+
+
+def get_user_default_selection(request, course_id=0):
+    logger.info(get_user_default_selection.__name__)
+    user_id = request.user.get_username()
+    default_view_type = request.GET.get('default_type')
+    key = 'default'
+    no_user_default_response = json.dumps({key: ''})
+    logger.info(f"the default option request from user {user_id} in course {course_id} of type: {default_view_type}")
```

Comment:
I think we'll need to reduce the log level of the messages in both of these methods.